modules/smart-agent_logstash/variables.tf

```diff
@@ -0,0 +1,375 @@
+  default     = {}
+}
+
+variable "events_in_low_aggregation_function" {
+  description = "Aggregation function and group by for events_in_low detector (i.e. \".mean(by=['host'])\")"
+  type        = string
+  default     = ""
+}
+
+variable "events_in_low_transformation_function" {
+  description = "Transformation function for events_in_low detector (i.e. \".mean(over='5m')\")"
+  type        = string
+  default     = ".mean(over='5m')"
+}
+
+variable "events_in_low_threshold_critical" {
+  description = "Critical threshold for events_in_low detector"
+  type        = number
+  default     = 0
+}
+
+variable "events_in_low_threshold_critical_lasting" {
+  description = "Critical threshold lasting for events_in_low detector"
+  type        = string
+  default     = "5m"
+}
+
+variable "events_in_low_threshold_major" {
+  description = "Major threshold for events_in_low detector"
+  type        = number
+  default     = 10
+}
+
+variable "events_in_low_threshold_major_lasting" {
+  description = "Major threshold lasting for events_in_low detector"
+  type        = string
+  default     = "10m"
+}
+
+# Events out
+variable "events_out_metric_name" {
+  description = "Metric name for input events"
+  type        = string
+  default     = "node.stats.events.events.out"
+}
+
+variable "events_out_high_tip" {
+  description = "Suggested first course of action or any note useful for incident handling"
+  type        = string
+  default     = ""
+}
+
+variable "events_out_high_runbook_url" {
+  description = "URL like SignalFx dashboard or wiki page which can help to troubleshoot the incident cause"
+  type        = string
+  default     = ""
+}
+
+variable "events_out_high_disabled" {
+  description = "Disable all alerting rules for events_out_high detector"
+  type        = bool
+  default     = null
+}
+
+variable "events_out_high_disabled_critical" {
+  description = "Disable critical alerting rule for events_out_high detector"
+  type        = bool
+  default     = null
+}
+
+variable "events_out_high_disabled_major" {
+  description = "Disable major alerting rule for events_out_high detector"
+  type        = bool
+  default     = null
+}
+
+variable "events_out_high_notifications" {
+  description = "Notification recipients list per severity overridden for events_out_high detector"
+  type        = map(list(string))
+  default     = {}
+}
+
+variable "events_out_high_aggregation_function" {
+  description = "Aggregation function and group by for events_out_high detector (i.e. \".mean(by=['host'])\")"
+  type        = string
+  default     = ""
+}
+
+variable "events_out_high_transformation_function" {
+  description = "Transformation function for events_out_high detector (i.e. \".mean(over='5m')\")"
+  type        = string
+  default     = ".mean(over='5m')"
+}
+
+variable "events_out_high_threshold_critical" {
+  description = "Critical threshold for events_out_high detector"
+  type        = number
+  default     = 30000
```

Comment:
may be some thresholds should not have default defined, this will force the user to set its own given that it is variable depending on each environment